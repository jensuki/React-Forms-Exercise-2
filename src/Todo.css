.Todo {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 10px auto;
    background: rgb(244, 244, 244);
    padding: 10px;
    width: 100%;
    max-width: 400px;
    border-radius: 5px;
}