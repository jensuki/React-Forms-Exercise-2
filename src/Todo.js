// should display a <div> with the task of the todo
import React from 'react';
import './Todo.css';

const Todo = ({ id, task, removeTodo }) => {
    const handleRemove = () => {
        removeTodo(id);
    }

    return (
        <div className="Todo">
            <p>{task}</p>
            <button className="Todo-remove" onClick={handleRemove}>X</button>
        </div>
    )
}

export default Todo;