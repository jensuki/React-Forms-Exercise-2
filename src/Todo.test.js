import { render, screen, fireEvent } from '@testing-library/react';
import Todo from './Todo';

it('renders without crashing', () => {
    render(<Todo id="1" task="Test Todo" removeTodo={() => { }} />);
});

it('matches snapshot', () => {
    const { asFragment } = render(<Todo id="1" task="Test Todo" removeTodo={() => { }} />);
    expect(asFragment()).toMatchSnapshot();
});

it('should display the todo task', () => {
    render(<Todo id="1" task="Test Todo" removeTodo={() => { }} />)

    expect(screen.getByText('Test Todo')).toBeInTheDocument();
})

it('calls removeTodo when X button is clicked', () => {
    const mockRemoveTodo = jest.fn();

    render(<Todo id="1" task="Test Todo" removeTodo={mockRemoveTodo} />)

    const removeButton = screen.getByText('X');

    fireEvent.click(removeButton);
    expect(mockRemoveTodo).toHaveBeenCalled();
})