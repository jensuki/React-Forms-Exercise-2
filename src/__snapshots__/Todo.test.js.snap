// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <p>
      Test Todo
    </p>
    <button
      class="Todo-remove"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
